src/internal/kopia/wrapper_scale_test.go

```diff
@@ -0,0 +1,168 @@
+package kopia
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/alcionai/clues"
+	"github.com/kopia/kopia/repo/manifest"
+	"github.com/kopia/kopia/snapshot"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/alcionai/corso/src/internal/data"
+	exchMock "github.com/alcionai/corso/src/internal/m365/service/exchange/mock"
+	"github.com/alcionai/corso/src/internal/tester"
+	"github.com/alcionai/corso/src/pkg/backup/identity"
+	"github.com/alcionai/corso/src/pkg/fault"
+	"github.com/alcionai/corso/src/pkg/path"
+)
+
+func BenchmarkHierarchyMerge(b *testing.B) {
+	ctx, flush := tester.NewContext(b)
+	defer flush()
+
+	c, err := openKopiaRepo(b, ctx)
+	require.NoError(b, err, clues.ToCore(err))
+
+	w := &Wrapper{c}
+
+	defer func() {
+		err := w.Close(ctx)
+		assert.NoError(b, err, clues.ToCore(err))
+	}()
+
+	var (
+		cols                 []data.BackupCollection
+		collectionLimit      = 1000
+		collectionItemsLimit = 3
+		itemData             = []byte("abcdefghijklmnopqrstuvwxyz")
+	)
+
+	baseStorePath, err := path.Build(
+		"a-tenant",
+		"a-user",
+		path.ExchangeService,
+		path.EmailCategory,
+		false,
+		"Inbox")
+	require.NoError(b, err, clues.ToCore(err))
+
+	for i := 0; i < collectionLimit; i++ {
+		folderName := fmt.Sprintf("folder%d", i)
+
+		storePath, err := baseStorePath.Append(false, folderName)
+		require.NoError(b, err, clues.ToCore(err))
+
+		col := exchMock.NewCollection(
+			storePath,
+			storePath,
+			collectionItemsLimit,
+		)
+
+		for j := 0; j < collectionItemsLimit; j++ {
+			itemName := fmt.Sprintf("item%d", j)
+			col.Names[j] = itemName
+			col.Data[j] = itemData
+		}
+
+		cols = append(cols, col)
+	}
+
+	reasons := []identity.Reasoner{
+		NewReason(
+			testTenant,
+			baseStorePath.ProtectedResource(),
+			baseStorePath.Service(),
+			baseStorePath.Category(),
+		),
```

Comment:
The new CI should flags this.

```suggestion
			baseStorePath.Category()),
```